### ISSUE TITLE
Should PoolStatModifier be able to modify the "value" of a PoolStat?

### ISSUE THREAD

**Issue by Abab-bk** [Zennyth/EnhancedStat#15]
The current PoolStatModifier can only modify the "max" and "min" attributes of a PoolStat, but it cannot modify the "value" directly. I understand that this may seem contradictory since "value" is not a stat. However, it is indeed a reasonable requirement. For example, let's say I have a buff that inflicts 30 points of damage to enemies every round. In this case, my health is a PoolStat, but the modifier cannot directly modify the "value" attribute; it can only modify "max" and "min".

You might argue that the correct approach is to make "health" a FloatStat, and "health_pool" should be the PoolStat. (I believe this is the correct approach.) However, in your documentation, you provided an example of a PoolStat using HealthStat, and PoolStat does have a "get_value()" method.
![image](https://github.com/Zennyth/EnhancedStat/assets/55276676/24bc254b-0863-4d76-b9c1-c01939bff2df)

**Comment by Zennyth** [comment]
Yeah I encountered the same problem when I started making the demo for https://github.com/Zennyth/EnhancedStat/issues/14.

I'll rework the way modifiers work in order to also apply an operation to value for raw values like PoolStat. I'll also add a way to implement your own tick logic.

I'll keep the issue open while I'm working on this.
